src/pyqasm/printer.py

```diff
@@ -0,0 +1,590 @@
+        ) from e
+
+    program.unroll()
+    program.remove_includes()
+
+    line_nums, sizes = _compute_line_nums(program)
+
+    global_phase = 0
+    statements: list[ast.Statement] = []
+
+    for s in program._statements:
+        if isinstance(s, ast.QuantumPhase):
+            global_phase += Qasm3ExprEvaluator.evaluate_expression(s.argument)[0]
+        else:
+            statements.append(s)
+
+    # Compute moments
+    moments, depths = _compute_moments(statements, line_nums)
+
+    if not idle_wires:
+        # remove all lines that are not used
+        ks = sorted(line_nums.keys(), key=lambda k: line_nums[k])
+        ks = [k for k in ks if depths[k] > 0]
+        line_nums = {k: i for i, k in enumerate(ks)}
+
+    fig = _mpl_draw(program, moments, line_nums, sizes, global_phase)
+
+    if filename is not None:
+        raise NotImplementedError("Saving to file not yet supported.")
+
+    return fig
+
+
+def _compute_line_nums(
+    module: QasmModule,
+) -> tuple[dict[tuple[str, int], int], dict[tuple[str, int], int]]:
+    """Compute line number and register size lookup table for the circuit."""
+    line_nums = {}
+    sizes = {}
+    line_num = -1
+    max_depth = 0
+
+    # Classical registers condensed to single line
+    for k in module._classical_registers:
+        line_num += 1
+        line_nums[(k, -1)] = line_num
+        sizes[(k, -1)] = module._classical_registers[k]
+
+    # Calculate qubit lines and depths
+    for qubit_reg in module._qubit_registers:
+        size = module._qubit_registers[qubit_reg]
+        line_num += size
+        for i in range(size):
+            line_nums[(qubit_reg, i)] = line_num
+            depth = module._qubit_depths[(qubit_reg, i)]._total_ops()
+            max_depth = max(max_depth, depth)
+            line_num -= 1
+        line_num += size
+
+    return line_nums, sizes
+
+
+# pylint: disable-next=too-many-branches
+def _compute_moments(
+    statements: list[ast.Statement], line_nums: dict[tuple[str, int], int]
+) -> tuple[list[list[QuantumStatement]], dict[tuple[str, int], int]]:
+    depths = {}
+    for k in line_nums:
+        depths[k] = -1
+
+    moments: list[list[QuantumStatement]] = []
+    for statement in statements:
+        if isinstance(statement, Declaration):
+            continue
+        if not isinstance(statement, QuantumStatement):
+            raise ValueError(f"Unsupported statement: {statement}")
+        if isinstance(statement, ast.QuantumGate):
+            qubits = [_identifier_to_key(q) for q in statement.qubits]
+            depth = 1 + max(depths[q] for q in qubits)
+            for q in qubits:
+                depths[q] = depth
+        elif isinstance(statement, ast.QuantumMeasurementStatement):
+            qubit_key = _identifier_to_key(statement.measure.qubit)
+            if statement.target is None:
+                raise NotImplementedError("Stand-alone measurement statements not yet supported.")
```

Comment:
Now that https://github.com/qBraid/pyqasm/pull/141 is merged can you also implement this?